plugin_tests/cli_results_test.py

```diff
@@ -64,28 +65,56 @@ def _runTest(self, cli_args=(), cli_kwargs={}, outputs={}, contains=[],
                     file.
                 'contains': a list of phrases that must be present in the first
                     256 kb of the file.
+        :param in_process: a flag indicating whether to run the cli in-process.
+            Defaults to True, and when True enables coverage testing to reach
+            the CLIs.  When False, subprocess is used, which enables capturing
+            stdout and stderr.
         :param contains: a list of phrases that must be present in the stdout
-            output of the cli.
+            output of the cli.  Only valid if in_process is False.
         :param excludes: a list of phrases that must be not present in the
-            stdout output of the cli.
+            stdout output of the cli.  Only valid if in_process is False.
         :returns: stdout from the test.
+
         """
         chunkSize = 256 * 1024
 
-        stdout = ''
+        if in_process and (contains or excludes):
+            raise ValueError("contains and excludes parameters may only be used"
+                             " if in_process is False.")
+
         cwd = os.environ.get('CLI_CWD')
-        cmd = ['python', os.environ['CLI_LIST_ENTRYPOINT']]
         tmppath = tempfile.mkdtemp()
         try:
-            cmd += [arg if not arg.startswith('tmp_') else
-                    os.path.join(tmppath, arg) for arg in cli_args]
+            cmd = [arg if not arg.startswith('tmp_') else
+                   os.path.join(tmppath, arg) for arg in cli_args]
             cmd += ['--%s=%s' % (
                 k, v if not v.startswith('tmp_') else os.path.join(tmppath, v))
                 for k, v in six.iteritems(cli_kwargs)]
-            process = subprocess.Popen(
-                cmd, shell=False, stdout=subprocess.PIPE, cwd=cwd)
-            stdout, stderr = process.communicate()
-            self.assertEqual(process.returncode, 0)
+            if in_process:
```

Comment:
My only concern is that, while this method does capture the output containing the desired strings, it doesn't work in the general case that something doesn't use the current value of `sys.stdout` or `sys.stderr` to determine where to write its output. Concretely, this doesn't capture the warning/error output from `large_image` when it tries to load a PNG as a TIFF.